### ISSUE TITLE
basePath support broken in 1.19.0

### ISSUE THREAD

**Issue by rrooding** [serverless-nextjs/serverless-next.js#877]
### Describe the bug
When using the basePath and trailingSlash settings in next.config.js, the lambda creates an incorrect url in serverless-component 1.19.0-alpha.27

### Actual behavior
With this config in next.config.js:
```
module.exports = {
  trailingSlash: true,
  basePath: "/nl"
  ...
}
```

When I do the following request:
`curl -v https://xxxxxxx.cloudfront.net/nl/zoek`

I get this response:
```
< HTTP/2 308
< content-length: 0
< server: CloudFront
< date: Mon, 18 Jan 2021 08:41:00 GMT
< location: /nl%2Fzoek/
< refresh: 0;url=/nl%2Fzoek/
```

### Expected behavior
When I do this response `curl -v https://xxxxxxx.cloudfront.net/nl/zoek` I expect the page `zoek.tsx` to be rendered.

### Steps to reproduce
Configure basePath & trailingSlash & deploy with serverless-component 1.19.0

### Versions
- @sls-next/serverless-component version: @sls-next/serverless-component@1.19.0-alpha.27
- Next.js version: 10.0.4

**Comment by dphang** [comment]
I think that is the expected behavior - if you set `trailingSlash: true` it means all URLs should only be accessible when the trailing slash is appended to the end of them. If you don't have it, it will instead return a 308 redirect response that redirects to the URL with the trailing slash. Note that CURL doesn't follow a redirect unless you add the `-L` flag. You should be able to confirm on the local Next.js dev server that this is the same behavior.

**Comment by rrooding** [comment]
I did a bit more research because I noticed I wasn't precise enough in the initial ticket, but I do notice a problem just when the app is deployed, and not locally. You are right that the redirect makes sense when fetching `/nl/zoek` as there is no trailing `/`. The problem however is the `%2F` in the redirect, and not a regular slash. 

The problem i'm experiencing in the browser is an endless loop if redirects. In curl this shows in the following way (with some response omitted):

Turned on follow redirects to demonstrate the browser problem:
```
$ curl -Lv https://xxxxxxx.cloudfront.net/nl/zoek
...
> GET /nl/zoek/ HTTP/2
< HTTP/2 308
< location: /nl%2Fzoek/
< refresh: 0;url=/nl%2Fzoek/
...
> GET /nl%2Fzoek/ HTTP/2
< HTTP/2 308
< location: /nl%2Fzoek
< refresh: 0;url=/nl%2Fzoek
...
> GET /nl%2Fzoek HTTP/2
< HTTP/2 308
< location: /nl%2Fzoek/
< refresh: 0;url=/nl%2Fzoek/
```

I hope this clarifies the problem a bit. Thanks for your response

**Comment by dphang** [comment]
I see, I couldn't repro the issue using the basepath app used in our e2e tests. The redirect works fine without a loop. Not sure where the encoded slash is coming from...

Do you have an example repo you could share that has this issue? Thanks!

**Comment by rrooding** [comment]
I will try to set one up today

**Comment by rrooding** [comment]
I've managed to reproduce it with an almost empty Next app. You can find it here: https://github.com/rrooding/sls-basepath-test

It's based on the same versions:
Next 10.0.4
Serverless-component 1.19.0-alpha.27

Only added `basePath` and `trailingSlash` to the next.config.js.

It's deployed here: https://d3t1363ttyedyi.cloudfront.net/nl/zoek

**Comment by dphang** [comment]
Ah, thanks, I think it seems to be the same issue as in: https://github.com/serverless-nextjs/serverless-next.js/pull/868. Because Next.js changed their internal redirect regex for Next 10.0.3, we needed to add it to our list to remove since we have our own redirect for trailing slash. But it seems the fix was incorrect - will fix it now. We need to remove these:

```json
  "redirects": [
    {
      "source": "/:file((?!\\.well-known(?:/.*)?)(?:[^/]+/)*[^/]+\\.\\w+)/",
      "destination": "/:file",
      "internal": true,
      "statusCode": 308,
      "regex": "^(?:/((?!\\.well-known(?:/.*)?)(?:[^/]+/)*[^/]+\\.\\w+))/$"
    },
    {
      "source": "/:notfile((?!\\.well-known(?:/.*)?)(?:[^/]+/)*[^/\\.]+)",
      "destination": "/:notfile/",
      "internal": true,
      "statusCode": 308,
      "regex": "^(?:/((?!\\.well-known(?:/.*)?)(?:[^/]+/)*[^/\\.]+))$"
    }
  ],
```

It looks like there is a new `internal` property added so we can easily identify it.

**Comment by dphang** [comment]
Fixed and published in `@sls-next/serverless-component@1.19.0-alpha.28`
